com_migratetojoomla/admin/src/controller/MainController.php

```diff
@@ -0,0 +1,278 @@
+     * @since  
+     */
+    protected $text_prefix = 'COM_MIGRATETOJOOMLA_MAIN';
+
+    /**
+     * Method to check media connection.
+     * 
+     * @since 1.0
+     */
+    public function checkMediaConnection()
+    {
+        $this->checkToken();
+
+        $app   = Factory::getApplication();
+
+        $data  = $this->input->post->get('jform', array(), 'array');
+
+        $this->testMediaConnection($data);
+        // Store data in session
+        $app->setUserState('com_migratetojoomla.main', $data);
+
+        // redirect in all case
+        $this->setRedirect(Route::_('index.php?option=com_migratetojoomla', false));
+    }
+
+    /**
+     * Controller funciton to check Database connection
+     * 
+     * @since 1.0
+     */
+    public function checkDatabaseConnection()
+    {
+        $this->checkToken();
+
+        $app   = Factory::getApplication();
+        $data  = $this->input->post->get('jform', array(), 'array');
+        $isconnected = MainHelper::testDatabaseConnection($data);
+
+        if ($isconnected) {
+            $app->enqueueMessage(TEXT::_('COM_MIGRATETOJOOMLA_DATABASE_CONNECTION_SUCCESSFULLY'), 'success');
+        } else {
+            $app->enqueueMessage(TEXT::_('COM_MIGRATETOJOOMLA_DATABASE_CONNECTION_UNSUCCESSFULLY'), 'danger');
+        }
+        // Store data in session
+        $app->setUserState('com_migratetojoomla.main', $data);
+
+        // redirect in all case
+        $this->setRedirect(Route::_('index.php?option=com_migratetojoomla', false));
+    }
+
+    /**
+     * Method to Download 
+     * 
+     * @since  1.0
+     */
+    public function download()
+    {
+        $this->checkMediaConnection();
+        $app   = Factory::getApplication();
+        try {
+
+            $data  = $this->input->post->get('jform', array(), 'array');
+            $load = new self();
+            $load->downloadMedia($data);
+
+            $app->enqueueMessage(TEXT::_('COM_MIGRATETOJOOMLA_DOWNLOAD_MEDIA_SUCCESSFULLY'), 'success');
+        } catch (\RuntimeException $th) {
+            $app->enqueueMessage($th->getMessage(), 'error');
+            // $app->enqueueMessage(TEXT::_('COM_MIGRATETOJOOMLA_DOWNLOAD__MEDIA_UNSUCCESSFULLY'), 'danger');
+
+        }
+        // redirect in all case
+        $this->setRedirect(Route::_('index.php?option=com_migratetojoomla', false));
+    }
+
+    /**
+     * Method to check connection with respective method
+     * 
+     * @param array form data
+     * @return boolean True on success
+     * 
+     * @since 1.0
+     */
+    public static function testMediaConnection($data = [])
+    {
+        $method = $data['mediaoptions'];
+
+        if ($method == 1) {
+            // Http
+            HttpHelper::testConnection($data['livewebsiteurl']);
+        } else if ($method == 2) {
+            // File system
+            FilesystemHelper::testConnection($data['basedir']);
+        } else if ($method == 3) {
+            FtpHelper::testConnection($data);
+        }
```

Comment:
Does not look like there is any "true" when successful...

Reply:
it will handle by testConnection method .